command/role.go

```diff
@@ -31,16 +30,43 @@ func role(ctx *Context, args []string) {
 		return
 	}
 
-	number, err := strconv.Atoi(args[1])
-	if err != nil {
-		ctx.ReportError("Invalid number", err)
-		return
-	} else if number < 0 || number >= len(ctx.Env.RoleColors) {
-		ctx.Reply("Invalid number")
-		return
+	type colorRole struct {
+		ID   string `json:"id"`
+		Name string `json:"color"`
+	}
+	var colorRoles []colorRole
+	for i, _ := range ctx.Env.RoleColors {
+		color, _ := ctx.Session.State.Role(ctx.Message.GuildID, ctx.Env.RoleColors[i])
+		name := fmt.Sprintf("%s", (color.Name))
+		r := colorRole{
+			ID:   color.ID,
+			Name: name,
+		}
+		colorRoles = append(colorRoles, r)
+
+	}
+	var roleID string
+	var addRole bool
+
+	for _, r := range colorRoles {
+		if args[1][0:1] == r.Name[0:1] || args[1] == r.Name {
```

Comment:
Prefix vs first char. Maybe make that more clear, or add actual prefix support, not sure if that's an issue.

Reply:
what do you mean? I don't get you